### PR TITLE
Fei2022 70 creatornetwork update

### PR DESCRIPTION
**Pull request by dorotanowak00**
Updated CreatorNetwork component:
-added time counter
-added websocket
-updated design to match the backend data

### CODE REVIEW

File: frontend/src/components/profileHorizontal/ProfileHorizontal.tsx [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
+    imageWidth,
+    imageUrl,
+    linkTo,
+}: Props) => {
+    const content = () => (
+        <div className={style.profile}>
+            <ProfilePicture width={imageWidth} url={imageUrl} />
+            <div className={style.userInfo}>
+                <p className={style.function}>{upperText ? upperText : ""}</p>
+                <p className={style.name}>{bottomText ? bottomText : ""}</p>
```
Comment (GabrLit) [inline frontend/src/components/profileHorizontal/ProfileHorizontal.tsx]: bottomText || ""

File: frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx [hunk 2]
```diff
@@ -1,67 +1,182 @@
...
+import { SocketContext } from "App";
+import { getAuction } from "API/UserService";
 
 export const CreatorNetwork = () => {
+    const socket = useContext(SocketContext);
+    const [auction, setAuction] = useState<AuctionItem>();
+    const [timeUntillAuctionEnds, setTimeUntillAuctionEnds] = useState<Duration>();
+    const [bidModalVisibility, setBidModalVisibility] = useState(false);
+    const lastBid = auction?.bidHistory[auction.bidHistory.length - 1]?.bid;
+    const user = useAppSelector((state) => state.user);
```
Comment (GabrLit) [inline frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx]: Refer to comment in marpuz PR for this one

File: frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx [hunk 3]
```diff
@@ -1,67 +1,182 @@
...
+                render: "Offer has to be higher than last bid",
+                type: "error",
+            });
+        if (user.userID === auction?.profileID.userID)
+            return toast.update(placeBidToast, {
+                render: "You cannot bid your own auction",
+                type: "error",
+                isLoading: false,
+                autoClose: 2500,
+                closeOnClick: true,
```
Comment (GabrLit) [inline frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx]: Refer to marpuz PR, move config to a wrapper. Code should be DRY! :D

**Review by GabrLit — COMMENTED** [review]
Some general structuring comments

File: frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx (hunk 2 above)
Comment (marpuz) [reply frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx]: I added selectors:

import { UserSelector } from "store/user";
const user = useAppSelector(UserSelector);

File: frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx (hunk 3 above)
Comment (marpuz) [reply frontend/src/views/HomePage/creatorNetwork/CreatorNetwork.tsx]: I made toast wrapper so you can use it as I do in my PR

**Pull request merged**